pkg/machines/libvirt-dbus.js

```diff
@@ -536,7 +536,7 @@ LIBVIRT_DBUS_PROVIDER = {
                         const network = parseNetDumpxml(xml);
                         dispatch(updateOrAddNetwork(Object.assign({}, props, network)));
                     })
-                    .catch(ex => console.warn('GET_NETWORK action failed failed for path', objPath, ex));
+                    .fail(ex => console.warn('GET_NETWORK action failed failed for path', objPath, ex));
```

Comment:
I always thought that .catch and .fail would do the same thing, and that .fail was just our old name for it. So apparently the difference is that `.fail` returns the original promise, but `.catch()` returns nothing? (i. e. for chaining)?

But `.fail()` isn't a standard [Promise API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). Could this perhaps use the two-function `.then()` for this purpose?

Reply:
I changed to using then, but I had to reject the exception again from inside the fail handler, since then will not propagate the original exception as fail does.

And to answer your first question, catch returns a new (resolved) promise, whereas fail returns the original promise. That's why I used the fail.